drs_cli/client.py

```diff
@@ -1,229 +1,424 @@
+        use_http: Set the URI schema of the DRS instance to `http` instead of
+            `https`when a DRS URI was provided to `uri`.
+        token: Bearer token to send along with DRS API requests. Set if
+            required by DRS implementation. Alternatively, specify in API
+            endpoint access methods.
 
     Attributes:
-        url: URL to DRS endpoints, composed of `host`, `port` and `base_path`,
-            e.g.,"https://my-drs.app:8080/ga4gh/drs/v1"
-        token: Bearer token for authentication.
+        uri: URI to DRS endpoints, built from `uri`, `port` and `base_path`,
+            e.g.,"https://my-drs.app:443/ga4gh/drs/v1".
+        token: Bearer token for gaining access to DRS endpoints.
         headers: Dictionary of request headers.
     """
+    # set regular expressions as private class variables
+    _RE_DOMAIN_PART = r'[a-z0-9]([a-z0-9-]{1,61}[a-z0-9])?'
+    _RE_DOMAIN = rf"({_RE_DOMAIN_PART}\.)+{_RE_DOMAIN_PART}\.?"
+    _RE_DRS_ID = r'.+'
+    _RE_HOST = rf"^(?P<schema>drs|http|https):\/\/(?P<host>{_RE_DOMAIN})\/?"
+    _RE_OBJECT_ID = rf"^(drs:\/\/{_RE_DOMAIN}\/)?(?P<obj_id>{_RE_DRS_ID})$"
 
     def __init__(
         self,
-        uri: str = 'http://0.0.0.0',
-        port: int = 443,
+        uri: str,
+        port: int = None,
         base_path: str = 'ga4gh/drs/v1',
+        use_http: bool = False,
         token: Optional[str] = None,
     ) -> Union[None, Error]:
-        
-        if check_drs_uri_regex(uri):
-            drs_host = get_drs_uri_host_and_id(uri)[0]
-            if port is self.__init__.__defaults__[1]:
-                self.url = f"https://{drs_host}/{base_path}"
-            else:
-                self.url = f"http://{drs_host}:{port}/{base_path}"
-        elif check_http_uri_regex(uri):
-            drs_host = get_http_uri_host(uri)
-            self.uri = f"http://{drs_host}:{port}/{base_path}"
-        elif check_https_uri_regex(uri):
-            drs_host = get_https_uri_host(uri)
-            self.uri = f"https://{drs_host}/{base_path}"
-        else:
-            return Error(**DRS_URI_FORMAT_ERROR)
+        """Class constructor."""
+        schema, host = self._get_host(uri)
+        if schema == 'drs':
+            schema = 'http' if use_http else 'https'
+        if port is None:
+            port = 80 if schema == 'http' else 443
+        self.uri = f"{schema}://{host}:{port}/{base_path}"
         self.token = token
-        self._get_headers()
-        logger.info(f"API URL: {self.url}")
+        self.headers = self._get_headers()
+        logger.info(f"Instantiated client for: {self.uri}")
 
     def get_object(
         self,
-        drs_object: str,
+        object_id: str,
         token: Optional[str] = None,
-    ) -> Union[Error, DrsObject]:
-        """Access DRS object.
+    ) -> Optional[Union[Error, DrsObject]]:
+        """Retrieve DRS object.
```

Comment:
Why is `Optional` added here?

Reply:
Good catch - it isn't! When I was working on the changes yesterday, I was switching between returning a Tuple of `(None, None)` and raising errors. Eventually I settled for raising errors, but I forgot to remove this, so please go ahead and remove it in your next commit.